### PR TITLE
feat: Add new identify method with time out support

### PR DESCRIPTION
**Pull request by keelerm84**
A callback provided to the `identify` method may not be executed for a
long time under the right circumstances (e.g. the network is slow or
unavailable).

Customers can now ensure their callbacks will fire within a specified
time interval by calling the new
`LDClient.identify(context:timeOut:completion:)` method.

### CODE REVIEW

**Review by louis-launchdarkly — APPROVED** [review]

File: LaunchDarkly/LaunchDarkly/LDClient.swift [hunk 1]
```diff
@@ -286,6 +286,38 @@ public class LDClient {
...
+            completion(true)
+        }
+
+        identify(context: context) {
+            guard !cancel else { return }
+
+            cancel = true
+            completion(false)
+        }
+    }
```
Comment (tanderson-ld) [inline LaunchDarkly/LaunchDarkly/LDClient.swift:319]: Is it acceptable that the timeout completion will be called with true, but the identify may still be internally processing and may even succeed with no successful completion callback?  I believe this case is possible.

Comment (tanderson-ld) [inline LaunchDarkly/LaunchDarkly/LDClient.swift:319]: Can you verify the following case, I'm unsure if this is an issue.

identify(context: myContextA, timeout: 10 seconds, completion: completionA)
identify(context: myContextB, timeout: 9.9 seconds, completion: completionB)

* assume identify of myContextA takes more than 0.1 seconds.

I believe completionA will be called with success, and completionB will be called with a timeout, but myContextB will be the active context.  I think the `internalIdentifyQueue` is involved here.

**Comment by tanderson-ld** [comment]
I think using @kinyoklion's 1 identify in progress spec may be the only course of action to add the timeout support on top of the existing code (which does not support cancellation of identify in progress).

**Comment by keelerm84** [comment]
Going to close this. Originally was thinking we might be able to make some concessions on the first round of this, but have decided against it.

**Pull request closed without merging**